### PR TITLE
Allow parts of previous migrations to be preserved

### PR DESCRIPTION
**Pull request by mrice32**
This PR adds command line options to allow certain portions of the system to be preserved in a call to `truffle migrate --reset`. The migration has been broken down into three components:
- The `Finder` contract.
- The `VoteToken` contract.
- Everything else. We may wish to break this down further at a later point, but until we have a use case in mind, it seemed easiest to keep the rest of the contracts bunched together.

I'm looking for feedback on these choices - I'm not sure they make sense from a usability perspective, so lmk what you think.

Issue #513.

### CODE REVIEW

**Comment by mrice32** [comment]
@ptare @rawfalafel PTAL.

File: common/MigrationUtils.js [hunk 1]
```diff
@@ -19,10 +27,76 @@ function shouldCommitDeployment(network) {
...
+  if (argList.length === 0) {
+    // No arguments, so add and return the empty txn options object.
+    argList.push(emptyTxnOptions);
+    return emptyTxnOptions;
+  } else {
+    // Determine if the last element is the txn options object.
+    const lastElement = argList[argList.length - 1];
+
+    if (
+      lastElement.from === undefined &&
```
Comment (rawfalafel) [inline common/MigrationUtils.js]: Obviously not your fault but this is bad API design, smh....

File: core/migrations/11_deploy_tokenized_derivative_creator.js [hunk 2]
```diff
@@ -3,32 +3,45 @@ const TokenizedDerivativeCreator = artifacts.require("TokenizedDerivativeCreator
...
 
   // Deploy whitelists.
-  const sponsorWhitelist = await deployAndGet(deployer, AddressWhitelist, { from: keys.sponsorWhitelist });
-  const returnCalculatorWhitelist = await deployAndGet(deployer, AddressWhitelist, {
-    from: keys.returnCalculatorWhitelist
-  });
-  const marginCurrencyWhitelist = await deployAndGet(deployer, AddressWhitelist, {
-    from: keys.marginCurrencyWhitelist
-  });
+  const { contract: sponsorWhitelist, didDeploy: deployedSponsorWhitelist } = await deploy(
```
Comment (rawfalafel) [inline core/migrations/11_deploy_tokenized_derivative_creator.js]: Looks like the `deployed*` consts aren't being used for anything?

Comment (mrice32) [reply core/migrations/11_deploy_tokenized_derivative_creator.js]: Nice catch - you're right, that was from an earlier iteration of the code that I forgot to delete.

File: common/MigrationUtils.js (hunk 1 above)
Comment (mrice32) [reply common/MigrationUtils.js]: Agreed :/

**Pull request merged**